ledger/src/blockstore.rs

```diff
@@ -1241,7 +1241,7 @@ impl Blockstore {
         metrics.commit_working_sets_elapsed_us += start.as_us();
 
         let mut start = Measure::start("Write Batch");
-        self.db.write(write_batch)?;
+        self.db.write_no_wal(write_batch)?;
```

Comment:
Weekend review so not digging super deep, but if I understand correctly, I believe we will never write anything to the WAL now. Using the `WriteBatch` ensures that we have "atomicity" across columns between writes. We have lots of assumptions of this atomicity in the codebase by way of `.unwrap()` and `.expect()`.

If we disable the WAL, the columns could easily get into an inconsistent state as the flushing of memtables is not synchronized. For example, imagine this hypothetical sequence:
- Some new shred inserted for slot `X` that fill slot `X`
- Insert fills up a `SlotMeta` memtable but not a data shred memtable
    - The `SlotMeta` for slot `X` may now get flushed
    - The data shreds for slot `X` still living only in memtable
- Node stops at some point after `SlotMeta` memtable has been flushed, everything in memtables (memory) now gone
- Node restarts, the `SlotMeta` for slot `X` that say shreds should be on disk, but they are not ==> panic

Disabling the WAL is something that has come up before and likely something that is necessary at some point, but we will need to adjust the rest of the codebase (this file + potentially users of `Blockstore`) to be more accommodating.